Kernel/FileSystem/VirtualFileSystem.h

```diff
@@ -118,8 +118,7 @@ class VirtualFileSystem {
     static bool check_matching_absolute_path_hierarchy(Custody const& first_custody, Custody const& second_custody);
     bool mount_point_exists_at_custody(Custody& mount_point);
 
-    // FIXME: This function is totally unsafe as someone could unmount the returned Mount underneath us.
-    Mount* find_mount_for_host_custody(Custody const& current_custody);
+    ErrorOr<void> find_and_do_on_mount_for_host_custody(Custody const& current_custody, Function<void(Mount&)>);
```

Comment:
The `find_and_do` sounds a bit weird, can we think of a better name? `apply_to_mount_for_host_custody` for example? Since we're not returning anything, the `find` part can be dropped anyway.